solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseCloudSolrClient.java

```diff
@@ -476,8 +478,11 @@ public void registerDocCollectionWatcher(String collection, DocCollectionWatcher
     assertZKStateProvider().zkStateReader.registerDocCollectionWatcher(collection, watcher);
   }
 
+  // TODO: async direct updates
```

Comment:
Before going further with this implementation, I wanted to get another opinion on the general implementation approach I'm taking here -- i.e. changing internal methods to return a CompletableFuture even for synchronous requests, except that they are already completed unless the request is async. A perhaps more hacky alternative is wrapping the CompletableFuture within a NamedList (similar to what I did [here](https://github.com/apache/lucene-solr/pull/1688)).

Reply:
Since we're merely talking about using a CompletableFuture wrapper (right?), it seems fine to me.  This is internal too.